packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -175,51 +175,57 @@ export class SOQLEditorInstance {
       }
       case MessageType.UI_TELEMETRY: {
         const { unsupported, errors } = e.payload as TelemetryModelJson;
-        if (errors) {
+        const hasErrors = Array.isArray(errors) ? errors.length : errors;
+        const hasUnsupported = Array.isArray(unsupported)
+          ? unsupported.length
+          : unsupported;
+        if (hasErrors) {
           trackError('syntax_error', JSON.stringify(e.payload)).catch(
             console.error
           );
         }
-        if (unsupported) {
+        if (hasUnsupported) {
           // no need to duplicate.  unsupported and errors often both present
-          if (!errors) {
+          if (!hasErrors) {
             trackError('syntax_unsupported', JSON.stringify(e.payload)).catch(
               console.error
             );
           }
-          channelService.appendLine(`This syntax is not yet supported.`);
+          channelService.appendLine(
+            `This syntax is not yet supported in SOQL Builder. Use a text editor.`
+          );
         }
         break;
       }
       case MessageType.SOBJECT_METADATA_REQUEST: {
-        this.retrieveSObject(e.payload as string).catch(async err => {
-          await showAndTrackError(MessageType.SOBJECT_METADATA_REQUEST, err);
+        this.retrieveSObject(e.payload as string).catch(err => {
+          showAndTrackError(MessageType.SOBJECT_METADATA_REQUEST, err);
           channelService.appendLine(
             `An error occurred while handling a request for object metadata for the ${e.payload} object.`
           );
         });
         break;
       }
       case MessageType.SOBJECTS_REQUEST: {
-        this.retrieveSObjects().catch(async err => {
-          await showAndTrackError(MessageType.SOBJECTS_REQUEST, err);
+        this.retrieveSObjects().catch(err => {
+          showAndTrackError(MessageType.SOBJECTS_REQUEST, err);
           channelService.appendLine(
             `An error occurred while handling a request for object names.`
           );
         });
         break;
       }
       case MessageType.RUN_SOQL_QUERY: {
-        this.handleRunQuery().catch(async err => {
-          await showAndTrackError(MessageType.RUN_SOQL_QUERY, err);
+        this.handleRunQuery().catch(err => {
+          showAndTrackError(MessageType.RUN_SOQL_QUERY, err);
           channelService.appendLine(
             `An error occurred while running the SOQL query.`
           );
         });
         break;
       }
       default: {
-        const message = `message type ${e.type} is not supported`;
+        const message = `Error: ${e.type}. This is an unknown error.  Open an issue: https://github.com/forcedotcom/soql-tooling/issues/new/choose.`;
```

Comment:
Emily - I simulated what this will look like in the channel output.  And confirmed the link is clickable.

<img width="1003" alt="Screen Shot 2020-12-08 at 10 37 08 AM" src="https://user-images.githubusercontent.com/599418/101520274-7ac01c00-3941-11eb-9c5b-20c14bc71896.png">

Reply:
Error: MESSAGE_UNSUPPORTED. Unknown error. Open an issue and supply the error message: <link>.